common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1315,6 +1315,7 @@ private void freeDeferredBufs() {
      */
```

Comment:
Wasn't able to comment on the `freeDeferredBufs()`  lines since it is not changed. We can change it to below to apply my other suggestions.
```
    /**
     * @return total number of deferred bytes
     */
    private long freeDeferredBufs() {
      if (deferredBufs == null || deferredBufs.isEmpty()) {
        deferredBufs = null;
        return 0;
      }
      long totalSize = 0;
      for (ByteBuffer deferredBuf : deferredBufs) {
        totalSize += deferredBuf.limit();
        mergeManager.pushMergeMetrics.deferredBlocks.mark(-1);
      }
      mergeManager.pushMergeMetrics.deferredBlockBytes.dec(totalSize);
      deferredBufs = null;
      return  totalSize;
    }
```

Reply:
This helps reduce iteration on `deferredBufs` which looks better. Have updated the PR, thanks for the suggestions.